### ISSUE TITLE
Files are never truncated

### ISSUE THREAD

**Issue by jcreedcmu** [jcreedcmu/twelf-mac#7]
Steps to reproduce:
- Open a file with TESample
- Change some of its contents, but make the new file shorter
- `File > Save`
- Close file and reopen it.
Observe:
- The content that was changed stays changed, but the content that should be truncated is still there.
 
It seems that whenever we save a file, even when we do
```
OSErr writeFile(TEHandle te, FSSpec *spec) {
  OSErr err;
  short refNum;
  err = FSpOpenDF(spec, fsRdWrPerm, &refNum);
  Handle textHandle = (Handle)TEGetText(te);
  long size = (*te)->teLength;
  err = SetFPos(refNum, fsFromStart, 0);
  err = FSWrite(refNum, &size, *textHandle);
  err = SetEOF(refNum, size);
  err = FSClose(refNum);
  return err;
}
```
the EOF of the file is not saved. [Inside Macintosh](https://dev.os9.ca/techpubs/mac/Files/Files-129.html#HEADING129-0) suggests we should call `FlushVol` after `FSClose`, but alas, `FlushVol` is not apparently supported by multiversal.

**Comment by agoode** [comment]
I think this might be a problem more generally in libretro. I noticed that `fopen("foo", "w");` was not properly truncating.

**Comment by agoode** [comment]
https://github.com/autc04/multiversal/pull/5

**Comment by jcreedcmu** [comment]
Fixed by above PR and 894f2be0c57f2f1a7e6981a4e979ad04197f7932 actually calling SetEOF.
